diff --git a/src/racing/actors.c b/src/racing/actors.c
index b835ad311..29cd29aee 100644
--- a/src/racing/actors.c
+++ b/src/racing/actors.c
@@ -650,7 +650,7 @@ void render_palm_trees(Camera* camera, Mat4 arg1) {
     Mat4 sp90;
     Vec3s sp88 = { 0, 0, 0 };
     s32 test;
-    size_t i = 0;
+    struct UnkActorSpawnData* const treeSpawnStart = var_s1;
 
     if (gGamestate == CREDITS_SEQUENCE) {
         var_f22 = 9000000.0f;
@@ -685,7 +685,7 @@ void render_palm_trees(Camera* camera, Mat4 arg1) {
             var_s1++;
             continue;
         }
-        FrameInterpolation_RecordOpenChild("render_palm_tree", TAG_ITEM_ADDR((i << 4) | (camera - cameras)));
+        FrameInterpolation_RecordOpenChild("render_palm_tree", TAG_ITEM_ADDR(((size_t) (var_s1 - treeSpawnStart) << 4) | (camera - cameras)));
 
         test &= 0xF;
         test = (s16) test;
@@ -727,7 +727,6 @@ void render_palm_trees(Camera* camera, Mat4 arg1) {
         }
         // @port Pop the transform id.
         FrameInterpolation_RecordCloseChild();
-        i++;
     }
 }
 
